### PR TITLE
Fix quite important AH bug

### LINKED ISSUE #687: Better auction left click error
### What happened?

When I left click on an item to inspect it in the auction house, the item stay in my hand and the menu don't open.

### Screenshots

image

### Log output

_No response_

### Minecraft Version

1.20.4

### Skyblocker Version

1.20.2+1.20.4

### Additional context

I found that the problem came from the "Jacob's ticket" item.
When I don't have it in my inventory, it work.
Someone had the same problem and he had a "Jacob's ticket" too in his inventory.

### PR DESCRIPTION
**Pull request by viciscat**
- fix #687 
- make fluboxer happy

### CODE REVIEW

File: src/main/java/de/hysky/skyblocker/skyblock/auction/AuctionViewScreen.java [hunk 1]
```diff
@@ -55,6 +57,15 @@ public AuctionViewScreen(AuctionHouseScreenHandler handler, PlayerInventory inve
         titleY = 4;
     }
 
+    @Override
+    public boolean keyPressed(int keyCode, int scanCode, int modifiers) {
+        if (keyCode == 256) {
```
Comment (AzureAaron) [inline src/main/java/de/hysky/skyblocker/skyblock/auction/AuctionViewScreen.java:62]: Use GLFW constant for clarity as to what key code this is

File: src/main/java/de/hysky/skyblocker/skyblock/auction/AuctionViewScreen.java [hunk 2]
```diff
@@ -189,7 +203,7 @@ public boolean mouseClicked(double mouseX, double mouseY, int button) {
 
     @Override
     public void onSlotChange(AuctionHouseScreenHandler handler, int slotId, ItemStack stack) {
-        if (stack.isOf(Items.BLACK_STAINED_GLASS_PANE) || slotId == 13) return;
+        if (stack.isOf(Items.BLACK_STAINED_GLASS_PANE) || slotId == 13 || slotId >= handler.getRows()*9) return;
```
Comment (AzureAaron) [inline src/main/java/de/hysky/skyblocker/skyblock/auction/AuctionViewScreen.java:206]: spacing in between the math

**Review by AzureAaron — CHANGES_REQUESTED** [review]

File: src/main/java/de/hysky/skyblocker/skyblock/auction/AuctionViewScreen.java [hunk 3]
```diff
@@ -78,10 +90,13 @@ protected void init() {
         verticalLayout.forEachChild(this::addDrawableChild);
         updateLayout();
 
-        addDrawableChild(new ButtonWidget.Builder(Text.literal("<"), button -> this.clickSlot(BACK_BUTTON_SLOT))
+        ButtonWidget build = new ButtonWidget.Builder(Text.literal("<"), button -> this.clickSlot(BACK_BUTTON_SLOT))
```
Comment (kevinthegreat1) [inline src/main/java/de/hysky/skyblocker/skyblock/auction/AuctionViewScreen.java:93]:
```suggestion
        ButtonWidget backButton = new ButtonWidget.Builder(Text.literal("<"), button -> this.clickSlot(BACK_BUTTON_SLOT))
```

File: src/main/java/de/hysky/skyblocker/skyblock/auction/AuctionViewScreen.java [hunk 4]
```diff
@@ -78,10 +90,13 @@ protected void init() {
...
                 .position(x + backgroundWidth - 16, y + 4)
                 .size(12, 12)
-                .build());
+                .tooltip(Tooltip.of(Text.literal("or press ESC!")))
+                .build();
+        build.setTooltipDelay(Duration.ofSeconds(2));
+        addDrawableChild(build);
```
Comment (kevinthegreat1) [inline src/main/java/de/hysky/skyblocker/skyblock/auction/AuctionViewScreen.java:99]:
```suggestion
        backButton.setTooltipDelay(Duration.ofSeconds(1));
        addDrawableChild(backButton);
```

**Review by kevinthegreat1 — CHANGES_REQUESTED** [review]

**Review by kevinthegreat1 — APPROVED** [review]
The slot change fix is basically a bodge but good enough. Also, I don't think we would have problems with spamming escape since it should close the screen, right?

**Comment by kevinthegreat1** [comment]
Squashing because commit history is trivial.

**Pull request merged**